File: SCT2/trunk/test-plugins/org.yakindu.sct.model.sexec.interpreter.test/test-gen/org/yakindu/sct/model/sexec/interpreter/test/AlwaysOncycleTest.java
/**
 * Copyright (c) 2013 committers of YAKINDU and others.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v10.html
 *
 * Contributors:
 *     committers of YAKINDU - initial API and implementation
 */
package org.yakindu.sct.model.sexec.interpreter.test;
import org.eclipse.xtext.junit4.InjectWith;
import org.eclipse.xtext.junit4.XtextRunner;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.yakindu.sct.model.sexec.interpreter.test.util.AbstractExecutionFlowTest;
import org.yakindu.sct.model.sexec.interpreter.test.util.SExecInjectionProvider;
import com.google.inject.Inject;
import org.junit.Before;
import org.yakindu.sct.model.sexec.interpreter.IExecutionFlowInterpreter;
import org.yakindu.sct.model.sexec.ExecutionFlow;
import org.yakindu.sct.test.models.SCTUnitTestModels;
import static org.junit.Assert.*;
/**
 *  Unit TestCase for AlwaysOncycle
 */
@SuppressWarnings("all")
@RunWith(XtextRunner.class)
@InjectWith(SExecInjectionProvider.class)
public class AlwaysOncycleTest extends AbstractExecutionFlowTest {

	@Inject
	private SCTUnitTestModels models;

	@Before
	public void setup() throws Exception {
		ExecutionFlow flow = models
				.loadExecutionFlowFromResource("AlwaysOncycle.sct");
		initInterpreter(flow);
	}
	@Test
	public void alwaysOncycleTest() throws Exception {
		interpreter.enter();
		assertTrue(isActive("StateA"));
		while (getInteger("value") < 5) {
			interpreter.runCycle();
			assertTrue(isActive("StateA"));
		}
		interpreter.runCycle();
		assertTrue(isActive("StateB"));
		while (getInteger("value") < 5) {
			interpreter.runCycle();
			assertTrue(isActive("StateB"));
		}
		interpreter.runCycle();
		assertTrue(isActive("StateA"));
	}
}

File: test-plugins/org.yakindu.sct.generator.c.test/gtests/AlwaysOncycleTest/AlwaysOncycleTest.cc
/**
* Copyright (c) 2017 committers of YAKINDU and others.
* All rights reserved. This program and the accompanying materials
* are made available under the terms of the Eclipse Public License v1.0
* which accompanies this distribution, and is available at
* http://www.eclipse.org/legal/epl-v10.html
*
* Contributors:
*     committers of YAKINDU - initial API and implementation
*/
#include "gtest/gtest.h"
#include "AlwaysOncycle.h"

AlwaysOncycle handle;

class StatemachineTest : public ::testing::Test{
	protected:
	virtual void SetUp() {
		alwaysOncycle_init(&handle);
	}
};


TEST_F(StatemachineTest, alwaysOncycleTest) {					
	alwaysOncycle_enter(&handle);
	EXPECT_TRUE(alwaysOncycle_isStateActive(&handle, AlwaysOncycle_main_region_StateA));
	while (alwaysOncycleIface_get_value(&handle)< 5l) {
		alwaysOncycle_runCycle(&handle);
		EXPECT_TRUE(alwaysOncycle_isStateActive(&handle, AlwaysOncycle_main_region_StateA));
	}
	alwaysOncycle_runCycle(&handle);
	EXPECT_TRUE(alwaysOncycle_isStateActive(&handle, AlwaysOncycle_main_region_StateB));
	while (alwaysOncycleIface_get_value(&handle)< 5l) {
		alwaysOncycle_runCycle(&handle);
		EXPECT_TRUE(alwaysOncycle_isStateActive(&handle, AlwaysOncycle_main_region_StateB));
	}
	alwaysOncycle_runCycle(&handle);
	EXPECT_TRUE(alwaysOncycle_isStateActive(&handle, AlwaysOncycle_main_region_StateA));
}


